pkg/util/cmapichecker/cmapichecker.go

```diff
@@ -99,34 +98,12 @@ func (o *cmapiChecker) Check(ctx context.Context) error {
 	}
 
 	if err := o.client.Create(ctx, cert); err != nil {
-		return &ApiCheckError{
-			SimpleError:     translateToSimpleError(err),
-			UnderlyingError: err,
-		}
+		return err
 	}
 	return nil
 }
 
-type ApiCheckError struct {
```

Comment:
Instead of returning an error with a "SimpleError" and an "UnderlyingError", we now only return the "UnderlyingError" and provide `TranslateToSimpleError` to obtain the "SimpleError" from this "UnderlyingError".